hooks/pre_validations/05-puppet_certs_exist.rb

```diff
@@ -0,0 +1,27 @@
+def error(message)
+  say message
+  logger.error message
+  kafo.class.exit 101
+end
+
+def param_value(mod, name)
+  param(mod, name).value if param(mod, name)
+end
+
+server_crl_path = param_value('foreman', 'server_ssl_crl')
+puppet_server_implementation = param_value('puppet', 'server_implementation')
+puppet_passenger = param_value('puppet', 'server_passenger')
+cert_dir = param_value('puppet', 'server_ssl_dir')
+puppet_ca_cert = File.join(cert_dir, 'ca/ca_crt.pem')
+puppet_enabled = kafo.module('puppet').enabled?
+
+client_message = "- is puppet already installed without puppet CA? You can remove the existing certificates with 'rm -rf #{cert_dir}' to get puppet CA properly configured."
+
+if cert_dir && File.directory?(cert_dir)
```

Comment:
The Puppet exec checks for existence of the SSL cert, not the dir, so it'd be best to mirror that.